resolwe/flow/managers/workload_connectors/slurm.py

```diff
@@ -0,0 +1,65 @@
+""".. Ignore pydocstyle D400.
+
+===============
+Slurm Connector
+===============
+
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+import os
+import shlex
+import subprocess
+
+from resolwe.utils import BraceMessage as __
+
+from .base import BaseConnector
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class Connector(BaseConnector):
+    """Slurm-based connector for job execution."""
+
+    def submit(self, data, dest_dir, argv, verbosity=1):
+        """Run process with SLURM.
+
+        For details, see
+        :meth:`~resolwe.flow.managers.workload_connectors.base.BaseConnector.submit`.
+        """
+        limits = data.process.get_resource_limits()
+        logger.debug(__(
+            "Connector '{}.{}' running {}.",
+            self.__class__.__module__,
+            self.__class__.__name__,
+            repr(argv)
+        ))
+        try:
+            script_path = os.path.join(dest_dir, 'slurm.sh')
+            with open(script_path, 'wt') as script:
+                script.write('#!/bin/bash\n')
+
+                # We add 200MB to the memory limit to account for
+                # executor overhead, since the executor is running in
+                # the same environment as the process.
+                script.write('#SBATCH --mem={}M\n'.format(limits['memory'] + 200))
+
+                script.write('#SBATCH --cpus-per-task={}\n'.format(limits['cores']))
+
+                # Render the argument vector into a command line.
+                line = ' '.join(map(shlex.quote, argv))
+                script.write(line + '\n')
+                os.chmod(script.fileno(), 0o555)
```

Comment:
Why is this done like this? The file should be created with the correct mode, not chmoded later.